src/StateFuture.ts

```diff
@@ -0,0 +1,312 @@
+export const execState: <S, E, A>(ma: StateFuture<S, E, A>, s: S) => F.Future<E, S> = T.execState
+
+/**
+ * @since 0.6.5
+ */
+export function left<S, E>(e: E): StateFuture<S, E, never> {
+  return fromFuture(F.left(e))
+}
+
+/**
+ * @since 0.6.5
+ */
+export const right: <S, A>(a: A) => StateFuture<S, never, A> = T.of
+
+/**
+ * @since 0.6.5
+ */
+export function rightTask<S, A>(ma: Task<A>): StateFuture<S, never, A> {
+  return fromFuture(F.rightTask(ma))
+}
+
+/**
+ * @since 0.6.5
+ */
+export function leftTask<S, E>(me: Task<E>): StateFuture<S, E, never> {
+  return fromFuture(F.leftTask(me))
+}
+
+/**
+ * @since 0.6.5
+ */
+export const fromFuture: <S, E, A>(ma: F.Future<E, A>) => StateFuture<S, E, A> = T.fromM
+
+/**
+ * @since 0.6.5
+ */
+export function rightIO<S, A>(ma: IO<A>): StateFuture<S, never, A> {
+  return fromFuture(F.rightIO(ma))
+}
+
+/**
+ * @since 0.6.5
+ */
+export function leftIO<S, E>(me: IO<E>): StateFuture<S, E, never> {
+  return fromFuture(F.leftIO(me))
+}
+
+/**
+ * @since 0.6.5
+ */
+export const rightState: <S, A>(ma: State<S, A>) => StateFuture<S, never, A> = T.fromState
+
+/**
+ * @since 0.6.5
+ */
+export function leftState<S, E>(me: State<S, E>): StateFuture<S, E, never> {
+  return s => F.left(me(s)[0])
+}
+
+/**
+ * @since 0.6.5
+ */
+export function orElse<S, E, M, A>(
+  f: (e: E) => StateFuture<S, M, A>
+): (ma: StateFuture<S, E, A>) => StateFuture<S, M, A> {
+  return ma => c =>
+    pipe(
+      ma(c),
+      F.orElse(e => f(e)(c))
+    )
+}
+
+/**
+ * @since 0.6.5
+ */
+export const get: <S>() => StateFuture<S, never, S> = T.get
+
+/**
+ * @since 0.6.5
+ */
+export const put: <S>(s: S) => StateFuture<S, never, void> = T.put
+
+/**
+ * @since 0.6.5
+ */
+export const modify: <S>(f: (s: S) => S) => StateFuture<S, never, void> = T.modify
+
+/**
+ * @since 0.6.5
+ */
+export const gets: <S, A>(f: (s: S) => A) => StateFuture<S, never, A> = T.gets
+
+/**
+ * @since 0.6.5
+ */
+export function fromEitherK<E, A extends Array<unknown>, B>(
+  f: (...a: A) => E.Either<E, B>
+): <S>(...a: A) => StateFuture<S, E, B> {
+  return (...a) => fromEither(f(...a))
+}
+
+/**
+ * @since 0.6.5
+ */
+export function chainEitherK<E, A, B>(
+  f: (a: A) => E.Either<E, B>
+): <S>(ma: StateFuture<S, E, A>) => StateFuture<S, E, B> {
+  return chain<any, E, A, B>(fromEitherK(f))
+}
+
+/**
+ * @since 0.6.5
+ */
+export function fromFutureK<E, A extends Array<unknown>, B>(
+  f: (...a: A) => F.Future<E, B>
+): <S>(...a: A) => StateFuture<S, E, B> {
+  return (...a) => fromFuture(f(...a))
+}
+
+/**
+ * @since 0.6.5
+ */
+export function chainFutureK<E, A, B>(
+  f: (a: A) => F.Future<E, B>
+): <S>(ma: StateFuture<S, E, A>) => StateFuture<S, E, B> {
+  return chain<any, E, A, B>(fromFutureK(f))
+}
+
+type MonadStateFuture = Monad3<URI> &
+  MonadThrow3<URI> &
+  Bifunctor3<URI> &
+  Alt3<URI> &
+  ChainRec3<URI> &
+  MonadThrow3<URI> &
+  MonadTask3<URI>
+
+/**
+ * @since 0.6.5
+ */
+export const stateFuture: MonadStateFuture = {
```

Comment:
please don't use a type alias here, otherwise the generated doc is a bit weird

```
export const stateFuture: MonadStateFuture = ...
```

Reply:
👍 added `"ter-indent": false` to tslint.json